app/components/dialogs/prioritize-build-modal.js

```diff
@@ -0,0 +1,104 @@
+import Component from '@ember/component';
+import { inject as service } from '@ember/service';
+import { computed } from '@ember/object';
+import eventually from 'travis/utils/eventually';
+import { reads } from '@ember/object/computed';
+import { task } from 'ember-concurrency';
+import {
+  bindKeyboardShortcuts,
+  unbindKeyboardShortcuts
+} from 'ember-keyboard-shortcuts';
+
+export default Component.extend({
+  keyboardShortcuts: {
+    'esc': 'toggleConfirmationModal'
+  },
+  flashes: service(),
+
+  didInsertElement() {
+    this._super(...arguments);
+    bindKeyboardShortcuts(this);
+  },
+
+  willDestroyElement() {
+    this._super(...arguments);
+    unbindKeyboardShortcuts(this);
+  },
+  isOpen: false,
+  options: computed(() => {
+    let arr = [
+      {
+        key: 'private1',
+        displayValue: 'item1',
+        cancelRunningJobsVal: false,
+        description: 'Place build at the top of the queue',
+        modalText: 'Place build at top of the queue',
+      },
+      {
+        key: 'private2',
+        displayValue: 'item2',
+        cancelRunningJobsVal: true,
+        description: 'Place build at the top of the queue and cancel all running jobs',
+        modalText: 'Place build at the top of the queue and cancel all running jobs',
+      }
+    ];
+    return arr;
+  }),
+
+  item: computed('type', 'job', 'build', function () {
+    let type = this.type;
+    let job = this.job;
+    let build = this.build;
+    if (type === 'job') {
+      return job;
+    } else {
+      return build;
+    }
+  }),
+
+  type: computed('job', 'build', function () {
+    let job = this.job;
+    if (job) {
+      return 'job';
+    } else {
+      return 'build';
+    }
+  }),
+
+  initialKey: '',
+  initial: computed('initialKey', 'options.@each.key', function () {
+    return this.options.findBy('key', this.initialKey);
+  }),
+  initialIndex: computed('initial', 'options.[]', function () {
+    return this.options.indexOf(this.initial);
+  }),
+
+  selectionKey: reads('initialKey'),
+  selection: computed('selectionKey', 'options.@each.key', function () {
+    return this.options.findBy('key', this.selectionKey);
+  }),
+  selectionIndex: computed('selection', 'options.[]', function () {
+    return this.options.indexOf(this.selection);
+  }),
+
+  increasePriorityTask: task(function* (value) {
+    this.set('isLoading', true);
+    yield eventually(this.item, (record) => {
```

Comment:
I don't think there's a need to use `eventually` here, because at the moment when this modal is shown, the build/job should already be resolved. What do you think?

Reply:
@artursmirnov : May you please let me know how I can interact with 'build' model in this case as in current scenario I am passing this.item in eventually function which is helping me to interact with model methods.

Reply:
@ShiviStar, sure! The only thing `eventually` does is it resolves a record if it has not been yet resolved. But _theoretically_ at the moment when this task is getting executed, the model should already be resolved, because it is rendered on current view. Therefore, `this.item` should contain already resolved record for either `build` or `job`, depending on current context. Although, you have the `increasePriority` method only for `build`, so you should treat them differently here: either `this.item.increasePriority()` for build or `this.item.build. increasePriority()` for job. 

Does it make sense?

Reply:
Done.